src/pricer/utils.ts

```diff
@@ -22,35 +23,19 @@ const tokenToApi: Record<string, ExchangeApiName[]> = {
   TKN: ['testExchange'],
 };
 
+const CACHE_EXPIRATION_TIME = 60_000;
 const INTERMEDIATE_CURRENCY = 'USD';
 
-type CoinBaseConstructorArgs = ConstructorParameters<typeof CoinBase>;
-type CoinCodexConstructorArgs = ConstructorParameters<typeof CoinCodex>;
-type CoinGeckoConstructorArgs = ConstructorParameters<typeof CoinGecko>;
-type ConstructorArgs =
-  | CoinBaseConstructorArgs
-  | CoinCodexConstructorArgs
-  | CoinGeckoConstructorArgs;
-
-const builders = new Map<
-  ExchangeApiName,
-  (args?: ConstructorArgs) => BaseExchangeApi
->();
-builders.set(
-  'coinBase',
-  (args: CoinBaseConstructorArgs = []) => new CoinBase(...args)
-);
-builders.set(
-  'coinCodex',
-  (args: CoinCodexConstructorArgs = []) => new CoinCodex(...args)
-);
+const builders = new Map<ExchangeApiName, ExchangeApi>();
+builders.set('coinBase', new CoinBase());
+builders.set('coinCodex', new CoinCodex());
 builders.set(
   'coinGecko',
-  (args: CoinGeckoConstructorArgs = []) => new CoinGecko(...args)
+  new CachedExchangeApi(new CoinGecko(), CACHE_EXPIRATION_TIME)
```

Comment:
should the CachedExchangeApi, applied to all of the apis?

Reply:
Yes, it doesn't hurt, but I use the same expiration time for all of them. See [this comment](https://github.com/rsksmart/rif-relay-client/pull/87#discussion_r1211643386).